app/services/visit_request/cancel.rb

```diff
@@ -6,10 +6,23 @@ def initialize(visit_request)
 
     def call
       visit_request.canceled!
+      attend_first_user_from_waitlist
     end
 
     private
 
     attr_reader :visit_request
+
+    def attend_first_user_from_waitlist
+      return unless waiting_list_request = VisitRequest.waiting_list.first
+
+      waiting_list_request.main_list!
+
+      if visit_request.event.registration?
+        VisitRequest::Approve.call(waiting_list_request)
+      else
+        VisitRequest::FinalConfirmation(waiting_list_request)
+      end
+    end
```

Comment:
As you see here -> https://codecov.io/gh/pivorakmeetup/pivorak-web-app/pull/417/src/app/services/visit_request/cancel.rb?before=app/services/visit_request/cancel.rb
This branch is not covered by specs.
You can add feature spec for an event with the different status.